ank/doc/swdesign/README.md

```diff
@@ -529,6 +529,223 @@ Needs:
+`swdd~cli-apply-accepts-ankaios-manifest-content-from-stdin~1`
+
+Status: approved
+
+**When** the user provides the manifest content via the CLI command `ank apply [OPTIONS] -` through `stdin`,
+
+**Then** the Ankaios CLI shall accept the given manifest content from `stdin`.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI provides a function to generate a state object from Ankaios manifests
+`swdd~cli-apply-generates-state-object-from-ankaios-manifests~1`
+
+Status: approved
+
+**Where** the user does not provide the optional argument `-d`,
+
+**When** the Ankaios CLI accepts the manifest content from file(s) or from `stdin`,
+
+**Then** the Ankaios CLI shall parse the manifest content into a state object.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI provides a function to generate filter masks from Ankaios manifests
+`swdd~cli-apply-generates-filter-masks-from-ankaios-manifests~1`
+
+Status: approved
+
+**When** the Ankaios CLI accepts the manifest content from file(s) or from `stdin`,
+
+**Then** the Ankaios CLI shall parse the manifest content into a list of filter masks.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI provides a function to send update state request for `ank apply ...`
+`swdd~cli-apply-send-update-state~1`
+
+Status: approved
+
+**When** the Ankaios CLI parses the manifest content into a state object
+
+**And** the Ankaios CLI parses the manifest content into a list of filter masks,
+
+**Then** then Ankaios CLI shall send an update state request to the Ankaios server containing the built state object and filter mask.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI provides a function to send update state request for `ank apply -d ...`
+`swdd~cli-apply-send-update-state-for-deletion~1`
+
+Status: approved
+
+**Where** the user provides the optional argument `-d`,
+
+**When** the Ankaios CLI parses the manifest content into a list of filter masks,
+
+**Then** then Ankaios CLI shall send an update state request to the Ankaios server containing an empty state object and the filter mask.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI provides a function to overwrite the agent names
+`swdd~cli-apply-ankaios-manifest-agent-name-overwrite~1`
+
+Status: approved
+
+**Where** the user provides the optional argument `--agent`,
+
+**When** the Ankaios CLI parses the manifest content into a state object,
+
+**Then** then Ankaios CLI shall overwrite the agent names in the state object, built as specified in the manifest content, with the one given by the argument.
+
+Needs:
+- impl
+- utest
+- stest
+
+#### CLI emits an error on absence of agent name
+`swdd~cli-apply-ankaios-manifest-error-on-agent-name-absence~1`
+
+Status: approved
+
+**If** the agent name is not specified in a workload specification
+
+**And** the user does not provide the agent name via the optional argument `--agent`,
+
+**When** the user runs the CLI command `ank apply [OPTIONS] ...`,
+
+**Then** the Ankaios CLI shall emit an agent name not specified error.
```

Comment:
I would start with the **When** sentence. I would also simplify the remaining condition, to "and at least one workload in the state object the Ankaios CLI should sender to the Ankaios server has no agent set".